ophyd/v2/core.py

```diff
@@ -570,11 +581,9 @@ def name(self) -> str:
     def set_name(self, name: str = ""):
         if name and not self._name:
             self._name = name
-            for attr_name, attr in self.__dict__.items():
-                # TODO: support lists and dicts of devices
-                if isinstance(attr, Device):
-                    attr.set_name(f"{name}-{attr_name.rstrip('_')}")
-                    attr.parent = self
+            for child_name, child in get_device_children(self):
+                child.set_name(f"{name}-{child_name}")
```

Comment:
This is where I'd restore the naming
```suggestion
                child.set_name(f"{name}-{child_name.rstrip('_')}")
```